src/classes/ADDR_Contact_TEST.cls

```diff
@@ -0,0 +1,1112 @@
+            contactsAddrs.contacts[i].MailingState = 'Washington';
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        update contactsAddrs.contacts;
+
+        for (Integer i = 0; i < contactsAddrs.contacts.size(); i++) {
+            contactsAddrs.contacts[i].MailingStreet = ' direct street  edit ';  // whitespace and casing should not count as a real edit.
+            contactsAddrs.contacts[i].MailingState = 'Oregon';
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        update contactsAddrs.contacts;
+
+        for (Integer i = 0; i < contactsAddrs.contacts.size(); i++) {
+            contactsAddrs.contacts[i].MailingStreet = ' DIRECT STREET  EDIT ';  // only change casing.
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.startTest();
+        update contactsAddrs.contacts;
+        Test.stopTest();
+        
+        map<Id, Contact> mapAccIdAcc = new map<Id, Contact>([select Id, Name, MailingStreet, MailingCity, MailingState, 
+        MailingPostalCode, MailingCountry, MailingLatitude, MailingLongitude, is_Address_Override__c, Current_Address__c from Contact]);
+        
+        for (Contact contact : mapAccIdAcc.values()) {
+            system.assert(contact.MailingStreet.equals('DIRECT STREET  EDIT'));
+            system.assertEquals(false, contact.is_Address_Override__c);
+            system.assertNotEquals(null, contact.Current_Address__c);
+        }
+ 
+        // verify that no new address was created
+        List<Address__c> ListAddr = [select Id, Default_Address__c, MailingStreet__c, Parent_Contact__c from Address__c];
+        system.assertEquals(2, ListAddr.size());
+        for (Address__c addr : ListAddr) {
+            system.assert(addr.MailingStreet__c.equals('DIRECT STREET  EDIT'));
+        }
+        
+        // now test that changing the case from the Contact updates all addresses
+        for (Integer i = 0; i < contactsAddrs.contacts.size(); i++) {
+            Contact contact = contactsAddrs.contacts[i];
+            contact.MailingStreet = 'direct street  edit';
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        update contactsAddrs.contacts;
+        
+        // verify that the Contact address
+        mapAccIdAcc = new map<Id, Contact>([select Id, Name, MailingStreet, MailingCity, MailingState, 
+        MailingPostalCode, MailingCountry, MailingLatitude, MailingLongitude, is_Address_Override__c, Current_Address__c from Contact]);
+        
+        for (Contact contact : mapAccIdAcc.values()) {
+            system.assert(contact.MailingStreet.equals('direct street  edit'));
+            system.assertEquals(false, contact.is_Address_Override__c);
+            system.assertNotEquals(null, contact.Current_Address__c);
+        }
+ 
+        // verify that no new address was created
+        ListAddr = [select Id, Default_Address__c, MailingStreet__c, Parent_Contact__c from Address__c];
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. Consider either introducing a `WHERE` clause, or adding a `LIMIT` to contain the number of records returned.